code/modules/mob/living/living_healthscan.dm

```diff
@@ -299,67 +300,106 @@ GLOBAL_LIST_INIT(known_implants, subtypesof(/obj/item/implant))
+					temp_advice = list(list(
 						"advice" = "Administer a single dose of dylovene.",
 						"icon" = "syringe",
 						"colour" = "green"
 						))
-				if((human_target_mob.getToxLoss() > 50 || (human_target_mob.getOxyLoss() > 50 && human_target_mob.blood_volume > 400) || human_target_mob.getBrainLoss() >= 10) && chemicals_lists["peridaxon"] < 5)
-					advice += list(list(
+					if(chemicals_lists["anti_toxin"])
+						if(chemicals_lists["anti_toxin"]["amount"] < 5)
+							advice += temp_advice
+					else
+						advice += temp_advice
+				if((human_target_mob.getToxLoss() > 50 || (human_target_mob.getOxyLoss() > 50 && human_target_mob.blood_volume > 400) || human_target_mob.getBrainLoss() >= 10))
+					temp_advice = list(list(
 						"advice" = "Administer a single dose of peridaxon.",
 						"icon" = "syringe",
 						"colour" = "grey"
 						))
-				if(human_target_mob.getOxyLoss() > 50 && chemicals_lists["dexalin"] < 5)
-					advice += list(list(
+					if(chemicals_lists["peridaxon"])
+						if(chemicals_lists["peridaxon"]["amount"] < 5)
+							advice += temp_advice
+					else
+						advice += temp_advice
+				if(human_target_mob.getOxyLoss() > 50)
+					temp_advice = list(list(
 						"advice" = "Administer a single dose of dexalin.",
 						"icon" = "syringe",
 						"colour" = "blue"
 						))
-				if(human_target_mob.getFireLoss(organic_only = TRUE) > 30 && (chemicals_lists["kelotane"] < 3 || chemicals_lists["dermaline"] < 3))
-					advice += list(list(
+					if(chemicals_lists["dexalin"])
+						if(chemicals_lists["dexalin"]["amount"] < 3)
+							advice += temp_advice
+					else
+						advice += temp_advice
+				if(human_target_mob.getFireLoss(organic_only = TRUE) > 30)
+					temp_advice = list(list(
 						"advice" = "Administer a single dose of kelotane.",
 						"icon" = "syringe",
 						"colour" = "yellow"
 						))
-				if(human_target_mob.getBruteLoss(organic_only = TRUE) > 30 && (chemicals_lists["bicaridine"] < 3 || chemicals_lists["meralyne"] < 3))
-					advice += list(list(
+					if(chemicals_lists["kelotane"])
+						if(chemicals_lists["kelotane"]["amount"] < 3)
+							advice += temp_advice
+					else
+						advice += temp_advice
+				if(human_target_mob.getBruteLoss(organic_only = TRUE) > 30)
+					temp_advice = list(list(
 						"advice" = "Administer a single dose of bicaridine.",
 						"icon" = "syringe",
 						"colour" = "red"
 						))
-				if(human_target_mob.health < 0 && chemicals_lists["inaprovaline"] < 5)
-					advice += list(list(
+					if(chemicals_lists["bicaridine"])
+						if(chemicals_lists["bicaridine"]["amount"] < 3)
+							advice += temp_advice
+					else
+						advice += temp_advice
+				if(human_target_mob.health < 0)
+					temp_advice = list(list(
 						"advice" = "Administer a single dose of inaprovaline.",
 						"icon" = "syringe",
 						"colour" = "purple"
 						))
-
+					if(chemicals_lists["inaprovaline"])
+						if(chemicals_lists["inaprovaline"]["amount"] < 5)
```

Comment:
can't you use ['od'] instead of hardcoding the od amounts? or use the same handling that the ['od'] variable takes for dynamically getting the OD amount?

Reply:
so, ["od"] is not what we are checking for, [5] or less is the amount that prompts recommending another dosage. _Idealy_ we could find a way to dynamically test for this using some sort of for loop setup. But it would take a complete refactor to do so.

However, to solve what you are asking for. we could have a ["recommend_dose"] variable that is gotten from the chemical. But so much of this is already hardcoded, it hardly makes a difference.